File: test/samples/emptylines/emptylines.cs
using System;

/// a char struct
struct StructWithAChar
{
    public char char1; //!< a char
    public int int2; //!< an integer
}

/// a struct
struct A
{
    public int[] listInStruct; //!< a list in a struct
    public StructWithAChar structInStruct; //!< a struct in a struct
}

/// a sized struct
struct SizedStruct
{
    public int size; //!< the size
    public string stringInStruct; //!< the string
}

class Program
{
    /// \param emptyList an empty list
    /// \param bufferString here to check correct parsing of empty line above
    /// \param n an integer, will be 0 in the sample input
    /// \param emptyInSample an empty list (only in the sample)
    /// \param emptyString an empty string
    /// \param main an other buffer string
    /// \param emptyCharList an empty char list
    /// \param nonEmptyCharList an char list, non empty
    /// \param structWithEmptyLine a struct containing an empty line, then a struct
    /// \param aSizedStruct a sized struct containing an empty line
    /// \param finish a string to finish
    static void EmptyLines(int[] emptyList, string bufferString, int n, int[] emptyInSample, string emptyString, string main, char[] emptyCharList, char[] nonEmptyCharList, A structWithEmptyLine, SizedStruct aSizedStruct, string finish)
    {
        /* TODO Wow, lots of empty lines! */
    }

    static void Main()
    {
        int[] emptyList = Array.ConvertAll(Console.ReadLine().Split(' ', StringSplitOptions.RemoveEmptyEntries), int.Parse);
        string bufferString = Console.ReadLine();
        int n = int.Parse(Console.ReadLine());
        int[] emptyInSample = Array.ConvertAll(Console.ReadLine().Split(' ', StringSplitOptions.RemoveEmptyEntries), int.Parse);
        string emptyString = Console.ReadLine();
        string main = Console.ReadLine();
        char[] emptyCharList = Console.ReadLine().ToCharArray();
        char[] nonEmptyCharList = Console.ReadLine().ToCharArray();
        A structWithEmptyLine;
        structWithEmptyLine.listInStruct = Array.ConvertAll(Console.ReadLine().Split(' ', StringSplitOptions.RemoveEmptyEntries), int.Parse);
        string[] words = Console.ReadLine().Split(' ');
        structWithEmptyLine.structInStruct = new StructWithAChar {char1 = words[0][0], int2 = int.Parse(words[1])};
        SizedStruct aSizedStruct;
        aSizedStruct.size = int.Parse(Console.ReadLine());
        aSizedStruct.stringInStruct = Console.ReadLine();
        string finish = Console.ReadLine();

        EmptyLines(emptyList, bufferString, n, emptyInSample, emptyString, main, emptyCharList, nonEmptyCharList, structWithEmptyLine, aSizedStruct, finish);
    }
}

File: test/samples/emptylines/emptylines.js
"use strict";
const fs = require("fs");


/**
 * @param {Array.<number>} emptyList an empty list
 * @param {string} bufferString here to check correct parsing of empty line above
 * @param {number} n an integer, will be 0 in the sample input
 * @param {Array.<number>} emptyInSample an empty list (only in the sample)
 * @param {string} emptyString an empty string
 * @param {string} main an other buffer string
 * @param {Array.<string>} emptyCharList an empty char list
 * @param {Array.<string>} nonEmptyCharList an char list, non empty
 * @param {Object} structWithEmptyLine a struct containing an empty line, then a struct
 * @param {Object} aSizedStruct a sized struct containing an empty line
 * @param {string} finish a string to finish
 * @returns {void}
 */
function emptyLines(emptyList, bufferString, n, emptyInSample, emptyString, main, emptyCharList, nonEmptyCharList, structWithEmptyLine, aSizedStruct, finish) {
    /* TODO Wow, lots of empty lines! */
}

function main(stdin) {
    let line = 0;

    const emptyList = stdin[line++].split(" ", 0).map(Number);
    const bufferString = stdin[line++];
    const n = Number(stdin[line++]);
    const emptyInSample = stdin[line++].split(" ", n).map(Number);
    const emptyString = stdin[line++];
    const main = stdin[line++];
    const emptyCharList = stdin[line++].split("");
    const nonEmptyCharList = stdin[line++].split("");
    const structWithEmptyLine = {};
    structWithEmptyLine.listInStruct = stdin[line++].split(" ", n).map(Number);
    const words = stdin[line++].split(" ");
    structWithEmptyLine.structInStruct = {
        char1: words[0],
        int2: Number(words[1])
    };
    const aSizedStruct = {};
    aSizedStruct.size = Number(stdin[line++]);
    aSizedStruct.stringInStruct = stdin[line++];
    const finish = stdin[line++];
    emptyLines(emptyList, bufferString, n, emptyInSample, emptyString, main, emptyCharList, nonEmptyCharList, structWithEmptyLine, aSizedStruct, finish);
}

let stdin = "";
process.stdin.on("data", data => stdin += data.toString())
             .on("end", () => main(stdin.split("\n")));
